src/sdk/endpoint_stats_collector.js

```diff
@@ -37,9 +37,9 @@ class EndpointStatsCollector {
         // clear this.send_stats to allow new updates to trigger another _send_stats
         this.send_stats = null;
         try {
-            await this.rpc_client.object.update_endpoint_stats(this.get_all_stats(), {
+            await P.all([await this.rpc_client.object.update_endpoint_stats(this.get_all_stats(), {
                 timeout: SEND_STATS_TIMEOUT
-            });
+            }), await this._send_nsfs_stats()]);
```

Comment:
I think it's better to change _send_stats to _send_endpoint_stats and to keep the old logic inside and in _send_stats just leave the P.all calling the 2 funcs, and the parts that we need for both calls.